BootloaderCommonPkg/Library/IppCryptoFipsLib/IppCryptoFipsLibE9.inf

```diff
@@ -0,0 +1,149 @@
+  $(IPP_PATH)/sources/ippcp/pcpbn.h
+  $(IPP_PATH)/sources/ippcp/pcptool.h
+  $(IPP_PATH)/sources/include/asmdefs.inc
+  $(IPP_PATH)/sources/include/ia_32e.inc
+  $(IPP_PATH)/sources/include/ia_common.inc
+  $(IPP_PATH)/sources/include/utils.inc
+
+  # Misc source files
+  $(IPP_PATH)/sources/ippcp/pcpmgf1ca_rmf.c
+  $(IPP_PATH)/sources/ippcp/gsmod_engineinit.c
+  $(IPP_PATH)/sources/ippcp/gsmod_enginegetsize.c
+  $(IPP_PATH)/sources/ippcp/gsmod_montfactor.c
+  $(IPP_PATH)/sources/ippcp/pcpngmontexpstuff_bin.c
+  $(IPP_PATH)/sources/ippcp/pcpmontred.c
+
+  # BigNum source files
+  $(IPP_PATH)/sources/ippcp/pcpbngetsize.c
+  $(IPP_PATH)/sources/ippcp/pcpbninit.c
+  $(IPP_PATH)/sources/ippcp/pcpbnsetoctstr.c
+  $(IPP_PATH)/sources/ippcp/pcpbngetoctstr.c
+  $(IPP_PATH)/sources/ippcp/pcpbnu_setoctstr.c
+  $(IPP_PATH)/sources/ippcp/pcpbnu_getoctstr.c
+  $(IPP_PATH)/sources/ippcp/pcpbnu_arith_mul_adc.c
+  $(IPP_PATH)/sources/ippcp/pcpbnu_arith_sqr_adc.c
+  $(IPP_PATH)/sources/ippcp/pcpbnu_arith_add.c
+  $(IPP_PATH)/sources/ippcp/pcpbnu32_arith_add.c
+  $(IPP_PATH)/sources/ippcp/pcpbnu_arith_addmuldigit.c
+  $(IPP_PATH)/sources/ippcp/pcpbnu_arith_sub.c
+  $(IPP_PATH)/sources/ippcp/pcpbnu32_arith_sub.c
+  $(IPP_PATH)/sources/ippcp/pcpbnu32_arith_submuldigit.c
+  $(IPP_PATH)/sources/ippcp/pcpbnu_nlz.c
+  $(IPP_PATH)/sources/ippcp/pcpbnu32_nlz.c
+  $(IPP_PATH)/sources/ippcp/pcpbnu32_arith_div.c
+
+  # RSA Source files
+  $(IPP_PATH)/sources/ippcp/pcprsa_getsizepublickey.c
+  $(IPP_PATH)/sources/ippcp/pcprsa_getbuffersizepublickey.c
+  $(IPP_PATH)/sources/ippcp/pcprsa_initpublickey.c
+  $(IPP_PATH)/sources/ippcp/pcprsa_setpublickey.c
+  $(IPP_PATH)/sources/ippcp/pcprsa_sizeof_pubkey.h
+  $(IPP_PATH)/sources/ippcp/pcprsaverifyhash_pkcs1v15_rmf.c
+  $(IPP_PATH)/sources/ippcp/pcprsaverify_pss_rmf.c
+  $(IPP_PATH)/sources/ippcp/pcprsa_gspub_cipher.c
+  $(IPP_PATH)/sources/ippcp/gsmodmethod_rsa.c
+  $(IPP_PATH)/sources/ippcp/pcprsa_montexpgetsize.c
+  $(IPP_PATH)/sources/ippcp/pcprsa_gsmethod__gpr_public.c
+
+  # Hash source files
+  $(IPP_PATH)/sources/ippcp/pcphashmessage_rmf.c
+  $(IPP_PATH)/sources/ippcp/pcphashinit_rmf.c
+  $(IPP_PATH)/sources/ippcp/pcphashupdate_rmf.c
+  $(IPP_PATH)/sources/ippcp/pcphashfinal_rmf.c
+  $(IPP_PATH)/sources/ippcp/pcphashca_rmf.c
+  $(IPP_PATH)/sources/ippcp/sm3/pcpsm3stuff.h
+
+  # HMAC source files
+  $(IPP_PATH)/sources/ippcp/pcphmacmessage_rmf.c
+  $(IPP_PATH)/sources/ippcp/pcphmacgetsize_rmf.c
+  $(IPP_PATH)/sources/ippcp/pcphmacca_rmf.c
+  $(IPP_PATH)/sources/ippcp/pcphmacupdate_rmf.c
+  $(IPP_PATH)/sources/ippcp/pcphmacfinal_rmf.c
+[Sources.IA32]
+
+
+[Sources.X64]
+  $(IPP_PATH)/sources/ippcp/asm_ia32/pcpvariant_xmm7560.inc
+  # E9 files
+  $(IPP_PATH)/sources/ippcp/asm_intel64/pcpsha256e9as.nasm
+  $(IPP_PATH)/sources/ippcp/asm_intel64/pcpsm3e9as.nasm
+  $(IPP_PATH)/sources/ippcp/asm_intel64/pcpsha512e9as.nasm
+  # M7 files
+  $(IPP_PATH)/sources/ippcp/asm_intel64/pcpbnuaddm7as.nasm
```

Comment:
WHat is the reason of having M7 ASM sources for every optimization inf's?

Reply:
I've added a readme file with some explanation of the different optimization levels.